frontend/src/routes/AdminRoute.tsx

```diff
@@ -12,8 +13,14 @@ const AdminRoute: React.FC<Props> = ({ children }: Props) => {
   const { currentUser } = useContext(AuthContext);
   const fetchCurrentUser = useAdminStore((state) => state.fetchCurrentUser);
   const user = useAdminStore((state) => state.user);
+  const [error, setError] = useState(false);
 
   useEffect(() => {
+    if (currentUser === null) {
+      setError(true);
+      return;
+    }
+    setError(false);
```

Comment:
you do not need to set the Error to false here, because there is an empty array set as dependency` []` in `useEffect` the hook will be called just once